IntelliTect.TestTools.WindowsTestWrapper/ExampleProjects/Notepad.AutomatedTests/Helpers/NotepadWindowHelpers.cs

```diff
@@ -10,6 +12,37 @@ public class NotepadWindowHelpers
         private readonly NotepadWindow _NotepadWindow = new NotepadWindow();
         private readonly SaveAsWindow _SaveAsWindow = new SaveAsWindow();
 
+        public bool ValidateTextFile(string location, string fileName, string expectedText)
+        {
+            bool comparisonPass = false;
+            string combinedPath = Path.Combine( location, fileName );
+            if ( combinedPath == expectedText )
+            {
+                comparisonPass = true;
+            }
+            return comparisonPass;
+        }
+
+        public void DeleteDocument(string location, string fileName)
+        {
+            var combinedPath = Path.Combine(location, fileName);
+            int retryAttempts = 0;
+            do
+            {
+                File.Delete(combinedPath);
+                if (!File.Exists(combinedPath))
+                {
+                    break;
+                }
+                Thread.Sleep(100);
+                retryAttempts++;
+            } while (retryAttempts < 10);
+            if (File.Exists(combinedPath))
```

Comment:
I was thinking of cases where (and I've seen this before,) the test tries to delete the file before it's finished saving. Would you recommend a very short Sleep instead? I was originally catching exceptions and retrying, but Mark pointed out that that's quite dangerous as it originally made LOTS of assumptions that no exception would ever be valid. I'm trying to find a happy medium between not catching exceptions, but also not having a method that will fail just because of some sub-second timing oddness!